src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -85,6 +85,31 @@ impl ScalarFunc {
         Ok(Some(i64::from(t.get_time().month())))
     }
 
+    #[inline]
+    pub fn year<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<i64>> {
+        let t = match self.children[0].eval_time(ctx, row) {
+            Err(err) => return Error::handle_invalid_time_error(ctx, err).map(|_| None),
+            Ok(None) => {
+                return Error::handle_invalid_time_error(
```

Comment:
I think there's no need calling `Error::handle_invalid_time_error`here . we can return ok(None) directly.
```sql
mysql> select year(null);
+------------+
| year(null) |
+------------+
|       NULL |
+------------+
1 row in set (0.00 sec)
```

No error or warning when input is `NULL`